src/base/memory_monitor_sr.cpp

```diff
@@ -169,6 +169,62 @@ namespace cubmem
 	  }
       }
   }
+
+  void memory_monitor::aggregate_server_info (MMON_SERVER_INFO &server_info)
+  {
+    strncpy (server_info.server_name, m_server_name.c_str (), m_server_name.size () + 1);
+    server_info.total_mem_usage = m_total_mem_usage.load ();
+    server_info.mmon_metainfo_mem_usage = m_meta_alloc_count * MMON_METAINFO_SIZE;
+    server_info.num_stat = m_tag_map.size ();
+
+    for (auto it = m_tag_map.begin (); it != m_tag_map.end (); ++it)
+      {
+	server_info.stat_info.emplace_back (it->first, m_stat_map[it->second].load ());
+      }
+
+    const auto &comp = [] (const auto &stat_pair1, const auto &stat_pair2)
+    {
+      return stat_pair1.second > stat_pair2.second;
+    };
+    std::sort (server_info.stat_info.begin (), server_info.stat_info.end (), comp);
+  }
+
+  void memory_monitor::finalize_dump ()
+  {
+    double mem_usage_ratio = 0.0;
+    FILE *outfile_fp = fopen ("finalize_dump.txt", "w+");
+    MMON_SERVER_INFO server_info;
+
+    auto MMON_CONVERT_TO_KB_SIZE = [] (uint64_t size)
+    {
+      return ((size) / 1024);
+    };
+
+    aggregate_server_info (server_info);
+
+    fprintf (outfile_fp, "====================cubrid memmon====================\n");
+    fprintf (outfile_fp, "Server Name: %s\n", server_info.server_name);
+    fprintf (outfile_fp, "Total Memory Usage(KB): %lu\n\n", MMON_CONVERT_TO_KB_SIZE (server_info.total_mem_usage));
+    fprintf (outfile_fp, "Total Metainfo Memory Usage(KB): %lu\n\n",
+	     MMON_CONVERT_TO_KB_SIZE (server_info.mmon_metainfo_mem_usage));
+    fprintf (outfile_fp, "-----------------------------------------------------\n");
+
+    fprintf (outfile_fp, "\t%-100s | %17s(%s)\n", "File Name", "Memory Usage", "Ratio");
+
+    for (const auto &s_info : server_info.stat_info)
```

Comment:
Nit.
I'm not sure it works for vector of pairs, but it does probably:
```suggestion
    for (const auto& [stat_name, mem_usage] : server_info.stat_info)
```